packages/components/dashboard/inventory-card/inventory-card.tsx

```diff
@@ -0,0 +1,388 @@
+    return (
+      <>
+        <RedExclamationCircleIcon className="popup-header-icon" />
+        {headerText}
+      </>
+    );
+  }
+  if (popUpType === PhaseType.PROCESSING) {
+    return (
+      <>
+        <BlueInProgressIcon className="popup-header-icon" />
+        {headerText}
+      </>
+    );
+  }
+  return <></>;
+};
+
+const CustomGalleryItem: React.FC<CustomGalleryItemProps> = ({
+  listItems,
+  listLoaded,
+  listError,
+  redirectPath,
+  resource,
+  statusMap,
+  errorPopup,
+  processingPopup,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <GalleryItem className="inventory-card-item">
+      {listLoaded && !listError ? (
+        <>
+          <div className="inventory-card-sub-item">
+            <Link to={redirectPath}>
+              {t('{{count}} {{resourceType}}', {
+                count: listItems.length,
+                resourceType: resource,
+              })}
+            </Link>
+          </div>
+          {statusMap && statusMap[PhaseType.ERROR] && (
+            <div className="icons-container">
+              <RedExclamationCircleIcon
+                title={PhaseType.ERROR}
+                className="icons"
+              />
+              {errorPopup}
+            </div>
+          )}
+          {statusMap && statusMap[PhaseType.PROCESSING] && (
+            <div>
+              <BlueInProgressIcon
+                title={PhaseType.PROCESSING}
+                className="icons"
+              />
+              {processingPopup}
+            </div>
+          )}
+        </>
+      ) : (
+        <Skeleton
+          screenreaderText={t('Loading {{resourceType}}', {
+            resourceType: resource,
+          })}
+        />
+      )}
+    </GalleryItem>
+  );
+};
+
+const BucketPopup: React.FC<PopupProps> = ({ type, statusmap, data }) => {
+  const { t } = useTranslation();
+  return (
+    <BucketClassPopOver
+      label={String(statusmap[type])}
+      bucketClasses={data as string[]}
+      headerContent={getHeaderHTMLElement(
+        type,
+        t('Buckets: {{type}}', {
+          type: type,
+        })
+      )}
+      trimContent
+    />
+  );
+};
+
+const DataResourcePopup: React.FC<PopupProps> = ({ type, statusmap, data }) => {
+  const { t } = useTranslation();
+  return (
+    <DataResourcesPopOver
+      label={String(statusmap[type])}
+      dataResources={data as string[]}
+      headerContent={getHeaderHTMLElement(
+        type,
+        t('Data sources: {{type}}', {
+          type: type,
+        })
+      )}
+      trimContent
+    />
+  );
+};
+
+const OBCPopup: React.FC<PopupProps> = ({ type, statusmap, data }) => {
+  const { t } = useTranslation();
+  return (
+    <OBCPopOver
+      label={String(statusmap[type])}
+      obcDetails={data as obcType[]}
+      headerContent={getHeaderHTMLElement(
+        type,
+        t('ObjectBucketClaims: {{type}}', {
+          type: type,
+        })
+      )}
+      trimContent
+    />
+  );
+};
```

Comment:
```suggestion

```
No need for these components...
call OBCPopOver and DataResourcesPopOver directly whereever needed...